### PR TITLE
feat: XE to OC System Clock

### PR DESCRIPTION
**Pull request by jrouliez**
PR for XE to OC System Clock

### CODE REVIEW

**Comment by jrouliez** [comment]
Closing PR

**Comment by jrouliez** [comment]
PR for XE to OC System Clock

File: package_nso_to_oc/xe/xe_system.py [hunk 1]
```diff
@@ -862,6 +864,31 @@ def cleanup_server_access(config_leftover, group_access_type, access_type):
...
         del config_leftover[f"tailf-ned-cisco-ios:{access_type}"]["server"]
 
+def xe_system_clock_timezone(config_before: dict, config_leftover: dict) -> None:
+    """
+    Translates NSO XE NED to MDD OpenConfig System Clock Timezone
+    """
+    openconfig_system_clock_config = openconfig_system["openconfig-system:system"][
+        "openconfig-system:clock"]["openconfig-system:config"]
+    timezone_name = ["UTC", "0", "0"] # Placeholder Timezone, Hours and Minutes
+
```
Comment (m0lass3s) [inline package_nso_to_oc/xe/xe_system.py:874]: Much of the repeated stuff can be stored in variables

```suggestion
    zone = config_before.get("tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("zone")
    hours = config_before.get("tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("hours")
    minutes = config_before.get("tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("minutes")
```

File: package_nso_to_oc/xe/xe_system.py [hunk 2]
```diff
@@ -862,6 +864,31 @@ def cleanup_server_access(config_leftover, group_access_type, access_type):
...
+        "openconfig-system:clock"]["openconfig-system:config"]
+    timezone_name = ["UTC", "0", "0"] # Placeholder Timezone, Hours and Minutes
+    
+    if config_before.get("tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("zone") and len(
+        config_before.get("tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("zone")) == 3:
+        timezone_name[0] = config_before.get("tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("zone")
+        del config_leftover["tailf-ned-cisco-ios:clock"]["timezone"]["zone"]
+
+    if config_before.get("tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("hours") or config_before.get(
+        "tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("hours") == 0:
```
Comment (m0lass3s) [inline package_nso_to_oc/xe/xe_system.py:881]: The if statement can then be simplified to...

```suggestion
    if hours != None:
```

File: package_nso_to_oc/xe/xe_system.py [hunk 3]
```diff
@@ -862,6 +864,31 @@ def cleanup_server_access(config_leftover, group_access_type, access_type):
...
+        timezone_name[1] = f'{config_before.get("tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("hours")}'
+        del config_leftover["tailf-ned-cisco-ios:clock"]["timezone"]["hours"]
+
+    if config_before.get("tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("minutes") or config_before.get(
+        "tailf-ned-cisco-ios:clock", {}).get("timezone", {}).get("minutes") == 0:
```
Comment (m0lass3s) [inline package_nso_to_oc/xe/xe_system.py:886]: This if statement can then be simplified to this...

```suggestion
    if minutes != None:
```

File: package_nso_to_oc/xe/xe_system.py (hunk 2 above)
Comment (m0lass3s) [inline package_nso_to_oc/xe/xe_system.py:881]: The if statement can then be simplified to...

```suggestion
    if hours != None:
```

File: package_nso_to_oc/xe/xe_system.py (hunk 3 above)
Comment (m0lass3s) [inline package_nso_to_oc/xe/xe_system.py:886]: This if statement can then be simplified to this...

```suggestion
    if minutes != None:
```

**Pull request merged**